src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -131,6 +154,32 @@ var rng = func() func(int) int {
 	}
 }()
 
+func (a *AdminAPI) mapBrokerIDsToURLs() {
+	err := a.eachBroker(func(aa *AdminAPI) error {
+		nc, err := aa.GetNodeConfig()
+		if err != nil {
+			return err
+		}
+		a.brokerIdToUrls.Store(nc.NodeID, aa.urls[0])
+		return nil
+	})
+	if err != nil {
+		log.Warn(fmt.Sprintf("failed to map brokerID to URL for 1 or more brokers: %v", err))
+	}
+}
+
+// GetLeaderID returns the broker ID of the leader of the Admin API
+func (a *AdminAPI) GetLeaderID() (*int, error) {
+	pa, err := a.GetPartition("redpanda", "controller", 0)
```

Comment:
Is the controller always to have just one partition?

Reply:
Yep, I don't think this'll change in the forseeable future.